### ISSUE TITLE
/etc/hosts not updated on disconnect

### ISSUE THREAD

**Issue by moxiegirl** [docker/docker#17342]
the network stack is updated but not `/etc/hosts` --- is this expected behavior?

```
 ubuntu@ip-172-31-36-118:~$ docker network disconnect isolated_nw container2
ubuntu@ip-172-31-36-118:~$ docker network disconnect bridge container2
ubuntu@ip-172-31-36-118:~$ docker attach container2

/ # cat /etc/hosts
172.17.0.3	8845961414ca
127.0.0.1	localhost
::1	localhost ip6-localhost ip6-loopback
fe00::0	ip6-localnet
ff00::0	ip6-mcastprefix
ff02::1	ip6-allnodes
ff02::2	ip6-allrouters
172.17.0.2	container1
172.17.0.2	container1.bridge
172.19.0.3	container3
172.19.0.3	container3.isolated_nw
/ # ubuntu@ip-172-31-36-118:~$ docker attach container2

/ # ifconfig
lo        Link encap:Local Loopback  
          inet addr:127.0.0.1  Mask:255.0.0.0
          inet6 addr: ::1/128 Scope:Host
          UP LOOPBACK RUNNING  MTU:65536  Metric:1
          RX packets:0 errors:0 dropped:0 overruns:0 frame:0
          TX packets:0 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:0 
          RX bytes:0 (0.0 B)  TX bytes:0 (0.0 B)

/ # cat /etc/hosts
172.17.0.3	8845961414ca
127.0.0.1	localhost
::1	localhost ip6-localhost ip6-loopback
fe00::0	ip6-localnet
ff00::0	ip6-mcastprefix
ff02::1	ip6-allnodes
ff02::2	ip6-allrouters
172.17.0.2	container1
172.17.0.2	container1.bridge
172.19.0.3	container3
172.19.0.3	container3.isolated_nw
/ # ping container1
PING container1 (172.17.0.2): 56 data bytes
ping: sendto: Network is unreachable
/ # 
```

**Comment by thaJeztah** [comment]
I think we had the same problem with `--link`, which tripped up some users. Would be happy to see this resolved though :+1:

**Comment by thaJeztah** [comment]
Similar issue for "rename"; https://github.com/docker/docker/issues/16485, and some older ones on entries for "removed" containers not being removed; https://github.com/docker/docker/issues/15811

**Comment by mavenugo** [comment]
this is resolved via https://github.com/docker/libnetwork/pull/709. will be available in docker/docker via next vendor-in.

**Comment by moxiegirl** [comment]
Closed by https://github.com/docker/libnetwork/pull/709

**Comment by aboch** [comment]
@moxiegirl 
Fix for this issue was ported to docker by #17381

**Comment by thaJeztah** [comment]
@aboch thanks, I'll close this one
